samples/bluetooth/central_hids/src/main.c

```diff
@@ -282,23 +282,27 @@ static void hids_c_ready_cb(struct bt_gatt_hids_c *hids_c)
 
 static void hids_on_ready(struct k_work *work)
 {
-	u8_t i;
 	int err;
+	struct bt_gatt_hids_c_rep_info *rep = NULL;
 
 	printk("HIDS is ready to work\n");
-	for (i = 0; i < hids_c.rep_cnt; ++i) {
-		struct bt_gatt_hids_c_rep_info *rep;
 
-		rep = hids_c.rep_info[i];
-		if (rep->ref.type ==
+	int fuse = 20;
+
+	while (NULL != (rep = bt_gatt_hids_c_rep_next(&hids_c, rep))) {
+		if (bt_gatt_hids_c_rep_type(rep) ==
 		    BT_GATT_HIDS_C_REPORT_TYPE_INPUT) {
 			printk("Subscribe in report id: %u\n",
-			       rep->ref.id);
+			       bt_gatt_hids_c_rep_id(rep));
 			err = bt_gatt_hids_c_rep_subscribe(&hids_c, rep,
 							   hids_c_notify_cb);
 			if (err) {
 				printk("Subscribe error (%d)\n", err);
 			}
+			if (--fuse == 0) {
```

Comment:
is this intended?

Reply:
ah.... no - it is an artefact from debugging. I will remove it in the moment.

Reply:
Done - good catch!

Reply:
:)